pkg/dataset/update_status.go

```diff
@@ -20,18 +20,18 @@ import (
 	"github.com/alluxio/k8s-operator/pkg/logger"
 )
 
-func UpdateDatasetStatus(ctx DatasetReconcilerReqCtx) (ctrl.Result, error) {
+func UpdateDatasetStatus(ctx *DatasetReconcilerReqCtx) (result ctrl.Result, err error) {
 	newestDataset := &alluxiov1alpha1.Dataset{}
-	if err := ctx.Get(ctx, ctx.NamespacedName, newestDataset); err != nil {
+	if err = ctx.Get(ctx, ctx.NamespacedName, newestDataset); err != nil {
 		logger.Errorf("Error getting newest dataset status: %v", err)
-		return ctrl.Result{}, err
+		return
 	}
-	if !reflect.DeepEqual(newestDataset.Status, ctx.Dataset.Status) {
-		newestDataset.Status = ctx.Dataset.Status
-		if err := ctx.Client.Status().Update(ctx.Context, newestDataset); err != nil {
+	if !reflect.DeepEqual(newestDataset.Status, ctx.Datasetter.GetStatus()) {
```

Comment:
i'm still not a fan of any unnecessary use of reflect. in this case, could the Status struct implement a `func (s *Status) Equals(other *Status) bool`

Reply:
Ooops, forgot this one

Reply:
No need to compare the whole `Status` anyways.